sdks/go/test/integration/primitives/checkpointing.go

```diff
@@ -57,7 +57,9 @@ func (fn *selfCheckpointingDoFn) SplitRestriction(_ []byte, rest offsetrange.Res
 	s := rest.Start
 	var splits []offsetrange.Restriction
 	for e := s + size; e <= rest.End; s, e = e, e+size {
-		splits = append(splits, offsetrange.Restriction{Start: s, End: e})
+		if s != e {
```

Comment:
Isn't it impossible for `s` to equal `e`? Won't `e` always equal `s + size`, with `size` always equaling 1?

Reply:
Yeah I'm having some trouble recreating the issue that was being described, and this isn't really doing anything.